### PR TITLE
Add GitHub OIDC module

### PR DESCRIPTION
**Pull request by spartan-haopham**
## Summary
Add GitHub OIDC centralization module

## Types of Changes

- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 🕷 Bug fix (non-breaking change which fixes an issue)
- [ ] 👏 Performance optimization (non-breaking change which addresses a performance issue)
- [ ] 🛠 Refactor (non-breaking change which does not change existing behavior or add new functionality)
- [ ] 📗 Library update (non-breaking change that will update one or more libraries to newer versions)
- [ ] 📝 Documentation (non-breaking change that doesn't change code behavior, can skip testing)
- [ ] ✅ Test (non-breaking change related to testing)
- [ ] 🔒 Security awareness (changes that effect permission scope, security scenarios)

## Test Plan
N/A

## Related Issues
https://www.notion.so/c0x12c/Stratos-15701fb05bf1803480e1d90a81eae2ab?pvs=4

### CODE REVIEW

File: aws/github-oidc/modules/provider/variables.tf [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  description = "List of client IDs (also known as audiences) for the IAM OIDC provider. Defaults to STS service if not values are provided"
+  type        = list(string)
+  default     = ["sts.amazonaws.com"]
+}
+
+variable "url" {
+  description = "The URL of the identity provider. Corresponds to the iss claim"
+  type        = string
+  default     = "https://token.actions.githubusercontent.com"
+}
```
Comment (thachanpy) [inline aws/github-oidc/modules/provider/variables.tf:21]: add EOL

File: aws/github-oidc/iam.tf [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        variable = condition.value.variable
+        values   = condition.value.values
+      }
+    }
+
+  }
+}
+
+resource "aws_iam_role" "github_actions_role" {
+  name               = "role-terraform-ops-${var.role_name}"
```
Comment (thachanpy) [inline aws/github-oidc/iam.tf:39]: please set the variable for this name

File: aws/github-oidc/examples/complete/outputs.tf [hunk 3]
```diff
@@ -0,0 +1,7 @@
+output "url" {
+  value = module.github_oidc.url
+}
+
+output "arn" {
+  value = module.github_oidc.arn
+}
```
Comment (thachanpy) [inline aws/github-oidc/examples/complete/outputs.tf:7]: EOL

**Review by spartan-toannhan — APPROVED** [review]
please resolve my comments before merging

**Review by thachanpy — APPROVED** [review]

**Pull request merged**